### ISSUE TITLE
Gmail CSV import failure on cyrillic strings

### ISSUE THREAD

**Issue by Pastafarianist** [owncloud/contacts#785]
I am trying to import my Google contacts to ownCloud. Here is an (edited, of course) example of what my CSV dump looks like ([version in original encoding](https://pastafarianist.tk/public.php?service=files&t=dd526f2ef30d5326b615a48f8d10920f)):

	Name,Given Name,Additional Name,Family Name,Yomi Name,Given Name Yomi,Additional Name Yomi,Family Name Yomi,Name Prefix,Name Suffix,Initials,Nickname,Short Name,Maiden Name,Birthday,Gender,Location,Billing Information,Directory Server,Mileage,Occupation,Hobby,Sensitivity,Priority,Subject,Notes,Group Membership,E-mail 1 - Type,E-mail 1 - Value,E-mail 2 - Type,E-mail 2 - Value,E-mail 3 - Type,E-mail 3 - Value,Phone 1 - Type,Phone 1 - Value,Phone 2 - Type,Phone 2 - Value,Phone 3 - Type,Phone 3 - Value,Phone 4 - Type,Phone 4 - Value,Phone 5 - Type,Phone 5 - Value,Website 1 - Type,Website 1 - Value
	Имя Фамилия,Имя,,Фамилия,,,,,,,,,,,1990-01-01,,,,,,,,,,,,Название группы,,,,,,,Home,+0 000 000-00-00,Mobile,+1 111 111-11-11,,,,,,,,
	Name Surname,Name,,Surname,,,,,,,,,,,,,,,,,,,,,,,* My Contacts ::: Group name 1 ::: Group name 2,Home,email1@example.com,* Home,email2@example.com,,,Home,+2 222 222-22-22,Mobile,+3 333 333-33-33,,,,,,,,

This is what my list of contacts looks like after importing:

![screenshot](http://i.imgur.com/zectZuJ.png)

The encoding is screwed. Furthermore, the second email of the second contact is not imported.

Which encoding does ownCloud expect to receive?

**Comment by babelouest** [comment]
the import expects a file encoded in UTF8, what is your import file incoding ?

**Comment by Pastafarianist** [comment]
@babelouest I did not modify the file that I exported from GMail (I have just re-checked that by re-exporting it again). Far Manager detects the encoding as UTF-16 LE.

**Comment by babelouest** [comment]
To avoid previous problems, all imported files are arbitrary converted into UTF-8 by the connector:
https://github.com/owncloud/contacts/blob/master/lib/connector/importcsvconnector.php#L62
So if GMail is exporting as UTF-16 LE, it will be converted to UTF-8 before import. Which probably means that during the conversion, special characters are not properly encoded.

If you manually convert your file from UTF-16 LE to UTF-8, with Notepad++ for example, do you have a proper conversion ?

**Comment by Pastafarianist** [comment]
Yes, if I manually convert it to UTF-8, there are no problems with encoding.

However, the import process freezes after 48/160 entries. After waiting for 20 minutes or so, I reloaded the page, and these 48 contacts were there, encoded properly. The rest didn't get imported, but I guess that's a different bug to report.

**Comment by babelouest** [comment]
I guess that's an existing bug from the encoding. Has the 49th contact some extra special characters ?

**Comment by Pastafarianist** [comment]
No. As a matter of fact, now I'm trying to import the same file but from a different computer. It freezes at 136/160.

**Comment by babelouest** [comment]
I think this has been resolved by now in the late updates, @Pastafarianist can you check it ?

**Comment by majki09** [comment]
I've had the same issue with my polish names. The Notepad++ detected UCS-2 Little Endian of google's file. However the conversion to UTF-8 with Notepad++ helps and everything's seems to be fine with names.
Could the conversion be done "over the air"?

**Comment by babelouest** [comment]
As I mentioned earlier, the conversion to UTF8 is made `over the air` by the import process, but the conversion is not full: https://github.com/fruux/sabre-vobject/wiki/Sabre-VObject-StringUtil

Maybe this will help: http://php.net/manual/function.mb-convert-encoding.php

I will make a pull request asap

**Comment by Pastafarianist** [comment]
@babelouest I have just tested it using ownCloud 8.2.2. The error was exactly as before: the original file I had exported from Google contacts produced unreadable characters, but after I converted it to UTF-8 manually ownCloud managed to import it successfully.
